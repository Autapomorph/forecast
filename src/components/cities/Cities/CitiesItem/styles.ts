import styled from 'styled-components/macro';

export const CitiesItemWrapper = styled.li`
  padding: 10px;
  margin-bottom: 15px;
  border: 1px solid var(--light-gray);
  border-radius: 5px;
  background: var(--white);
  color: var(--contrast-text-color);
  font-weight: bold;

  :last-child {
    margin-bottom: 0;
  }
`;

export const CitiesItemTitle = styled.span`
  cursor: pointer;
`;

export const TitleButtonsWrapper = styled.div`
  flex-shrink: 0;

  > :not(:last-child) {
    margin-right: 10px;
  }
`;

export const CitiesItemHeader = styled.h4`
  display: flex;
  align-items: center;
  margin: 5px 0;
  font-weight: bold;

  ${CitiesItemTitle} {
    flex-grow: 1;
  }
`;

export const CitiesItemContent = styled.p`
  margin: 10px 0 0 0;
`;
