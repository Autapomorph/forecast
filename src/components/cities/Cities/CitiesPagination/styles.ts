import styled, { css } from 'styled-components/macro';

type Props = {
  isSinglePage: boolean;
};

export const CitiesPagination = styled.div`
  .pagination {
    display: flex;
    justify-content: flex-end;
    list-style: none;

    ${({ isSinglePage }: Props) =>
      isSinglePage &&
      css`
        display: none;
      `}
  }

  .page {
    background: var(--white);
    color: var(--contrast-text-color);
    transition: 0.3s;
    cursor: pointer;

    :hover {
      background: var(--gray);
    }

    &--prev {
      border-radius: 5px 0 0 5px;
    }

    &--next {
      border-radius: 0 5px 5px 0;
    }

    &--disabled {
      background: var(--light-gray);
      color: var(--disabled-text-color);
      cursor: default;

      :hover {
        background: var(--light-gray);
        color: var(--disabled-text-color);
      }
    }

    &--hidden {
      display: none;
    }

    a {
      display: block;
      padding: 10px;
      outline: none;
    }
  }
`;
