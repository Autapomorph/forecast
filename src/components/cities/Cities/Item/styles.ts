import styled from 'styled-components/macro';

import { ThemeProp } from 'models';

export const CitiesItemWrapper = styled.li`
  display: flex;
  min-height: 50px;
  margin-bottom: 15px;
  border: 1px solid ${({ theme }: ThemeProp) => theme.borderColor};
  border-radius: 5px;
  background: ${({ theme }: ThemeProp) => theme.elementBg};
  color: ${({ theme }: ThemeProp) => theme.altTextColor};
  font-weight: bold;

  :last-child {
    margin-bottom: 0;
  }
`;

export const CitiesItemTitle = styled.span`
  align-self: center;
  flex-grow: 1;
  padding: 15px 0;
  padding-left: 10px;
  cursor: pointer;
`;

export const TitleButtonsWrapper = styled.div`
  display: flex;
  flex-shrink: 0;
  padding-left: 10px;
  padding-right: 5px;

  > :not(:last-child) {
    margin-right: 10px;
  }
`;
