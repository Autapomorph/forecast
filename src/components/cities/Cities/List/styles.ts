import styled from 'styled-components/macro';

export const CitiesList = styled.ul`
  padding: 0;
  margin: 12px 0 0;
  list-style: none;
`;
