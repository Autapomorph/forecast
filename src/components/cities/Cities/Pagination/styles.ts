import styled, { css } from 'styled-components/macro';

import { ThemeProp } from 'models';

type Props = {
  isSinglePage: boolean;
};

export const CitiesPagination = styled.div`
  .pagination {
    display: flex;
    justify-content: flex-end;
    list-style: none;

    ${({ isSinglePage }: Props) =>
      isSinglePage &&
      css`
        display: none;
      `}
  }

  .page {
    background: ${({ theme }: ThemeProp) => theme.paginationBg};
    color: ${({ theme }: ThemeProp) => theme.paginationColor};
    transition: 0.3s;
    cursor: pointer;

    :hover {
      background: ${({ theme }: ThemeProp) => theme.paginationHoverBg};
    }

    &--prev {
      border-radius: 5px 0 0 5px;
    }

    &--next {
      border-radius: 0 5px 5px 0;
    }

    &--disabled {
      background: ${({ theme }: ThemeProp) => theme.paginationDisabledBg};
      color: ${({ theme }: ThemeProp) => theme.paginationDisabledColor};
      cursor: default;

      :hover {
        background: ${({ theme }: ThemeProp) => theme.paginationDisabledBg};
        color: ${({ theme }: ThemeProp) => theme.paginationDisabledColor};
      }
    }

    &--hidden {
      display: none;
    }

    a {
      display: block;
      padding: 10px;
      outline: none;
    }
  }
`;
