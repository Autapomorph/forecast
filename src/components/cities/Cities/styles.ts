import styled from 'styled-components/macro';

type Props = {
  isLoading: boolean;
};

export const CitiesSection = styled.section`
  grid-area: cities;
  padding: 0 10px;
  margin-bottom: ${({ isLoading }: Props) => (isLoading ? '0' : '20px')};
`;

export const CitiesHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
