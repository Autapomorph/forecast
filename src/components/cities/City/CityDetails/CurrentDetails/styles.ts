import styled from 'styled-components/macro';

export const CurrentDetailsWrapper = styled.div`
  grid-area: city-current;
  margin-top: 12px;

  @media screen and (min-width: 600px) {
    margin-top: 0;
  }
`;

export const Item = styled.dl`
  display: grid;
  grid-template-columns: [wd-title] fit-content(50%) [wd-desc] auto;
  grid-gap: 5px 20px;
  align-items: baseline;
`;

export const ItemTitle = styled.dt`
  grid-column: wd-title;
  margin: 0;
`;

export const ItemDescription = styled.dd`
  grid-column: wd-desc;
  margin: 0;
`;
