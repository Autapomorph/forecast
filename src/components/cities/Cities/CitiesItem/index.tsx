import React from 'react';

import { City, Coords } from 'models';
import generateName from 'utils/city/generateName';
import FeaturedButton from 'components/common/buttons/FeaturedButton';
import CountryFlag from 'components/common/icons/CountryFlag';

import * as S from './styles';

type Props = {
  city: City;
  isFeatured: boolean;
  fetchCity: (cityCoords: Coords) => void;
  addToFeatured: (city: City) => void;
  removeFromFeatured: (cityId: City['id']) => void;
};

const CitiesItem = ({
  city,
  isFeatured,
  fetchCity,
  addToFeatured,
  removeFromFeatured,
}: Props): React.ReactElement => (
  <S.CitiesItemWrapper>
    <S.CitiesItemHeader>
      <S.CitiesItemTitle onClick={() => fetchCity(city.coords)}>
        <span style={{ marginRight: '0.3em' }}>{`${generateName(city)}`}</span>
        <CountryFlag country={city.countryCode.toLowerCase()} size="1.2em" />
      </S.CitiesItemTitle>

      <S.TitleButtonsWrapper>
        <FeaturedButton
          isFeatured={isFeatured}
          onRemove={() => removeFromFeatured(city.id)}
          onAdd={() =>
            addToFeatured({
              id: city.id,
              name: city.name,
              region: city.region,
              countryName: city.countryName,
              countryCode: city.countryCode,
              coords: city.coords,
            })
          }
        />
      </S.TitleButtonsWrapper>
    </S.CitiesItemHeader>
  </S.CitiesItemWrapper>
);

export default CitiesItem;
