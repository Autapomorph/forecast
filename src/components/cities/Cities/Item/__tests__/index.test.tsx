import React from 'react';
import { shallow } from 'enzyme';

import CitiesItem from '..';

import * as S from '../styles';

describe('CitiesItem tests', () => {
  const props = {
    city: {
      id: 0,
      name: 'name',
      region: 'region',
      countryName: 'countryName',
      countryCode: 'countryCode',
      coords: { latitude: 0, longitude: 0 },
    },
    isFeatured: false,
    fetchCity: jest.fn(),
    addToFeatured: jest.fn(),
    removeFromFeatured: jest.fn(),
  };

  it('renders properly', () => {
    const wrapper = shallow(<CitiesItem {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchCity on CitiesItemTitle click', () => {
    const wrapper = shallow(<CitiesItem {...props} />);
    wrapper.find(S.CitiesItemTitle).invoke('onClick')();

    expect(props.fetchCity).toBeCalledWith(props.city.coords);
  });

  it('should call addToFeatured on FeaturedButton onAdd callback', () => {
    const wrapper = shallow(<CitiesItem {...props} />);
    // @ts-ignore
    wrapper.find('FeaturedButton').invoke('onAdd')();

    const {
      city: { id, name, region, countryName, countryCode, coords },
    } = props;
    expect(props.addToFeatured).toBeCalledWith({
      id,
      name,
      region,
      countryName,
      countryCode,
      coords,
    });
  });

  it('should call removeFromFeatured on FeaturedButton onRemove callback', () => {
    const wrapper = shallow(<CitiesItem {...props} />);
    // @ts-ignore
    wrapper.find('FeaturedButton').invoke('onRemove')();

    expect(props.removeFromFeatured).toBeCalledWith(props.city.id);
  });
});
