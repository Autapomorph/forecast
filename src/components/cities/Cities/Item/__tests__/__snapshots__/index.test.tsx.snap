// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesItem tests renders properly 1`] = `
<styles__CitiesItemWrapper>
  <styles__CitiesItemTitle
    onClick={[Function]}
  >
    <span>
      name, region, 
    </span>
    <span
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      countryName
      <MarginRight__MarginRightDelimeter />
      <CountryFlag
        country="countrycode"
      />
    </span>
  </styles__CitiesItemTitle>
  <styles__TitleButtonsWrapper>
    <FeaturedButton
      isFeatured={false}
      onAdd={[Function]}
      onRemove={[Function]}
    />
  </styles__TitleButtonsWrapper>
</styles__CitiesItemWrapper>
`;
